### ISSUE TITLE
Build error against PostgreSQL 14

### ISSUE THREAD

**Issue by devrimgunduz** [darold/pgtt#19]
Hi,

pgtt 2.5 fails to build against v14. Log is below. Can you please take a look? v14 is around the corner.

> gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -I/usr/pgsql-14/include -Wno-uninitialized -I. -I./ -I/usr/pgsql-14/include/server -I/usr/pgsql-14/include/internal  -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include  -c -o pgtt.o pgtt.c
/usr/lib64/ccache/clang -Wno-ignored-attributes -fno-strict-aliasing -fwrapv -O2  -I/usr/pgsql-14/include -Wno-uninitialized -I. -I./ -I/usr/pgsql-14/include/server -I/usr/pgsql-14/include/internal  -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include -flto=thin -emit-llvm -c -o pgtt.bc pgtt.c
pgtt.c: In function '_PG_init':
pgtt.c:304:29: warning: assignment to 'ProcessUtility_hook_type' {aka 'void (*)(PlannedStmt *, const char *, _Bool,  ProcessUtilityContext,  struct ParamListInfoData *, QueryEnvironment *, struct _DestReceiver *, QueryCompletion *)'} from incompatible pointer type 'void (*)(PlannedStmt *, const char *, ProcessUtilityContext,  struct ParamListInfoData *, QueryEnvironment *, DestReceiver *, QueryCompletion *)' {aka 'void (*)(PlannedStmt *, const char *, ProcessUtilityContext,  struct ParamListInfoData *, QueryEnvironment *, struct _DestReceiver *, QueryCompletion *)'} [-Wincompatible-pointer-types]
  304 |         ProcessUtility_hook = gtt_ProcessUtility;
      |                             ^
pgtt.c: In function 'gtt_ProcessUtility':
pgtt.c:98:62: error: incompatible type for argument 4 of 'prev_ProcessUtility'
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                              ^~~~~~
      |                                                              |
      |                                                              ParamListInfo {aka struct ParamListInfoData *}
pgtt.c:373:45: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c:98:62: note: expected 'ProcessUtilityContext' but argument is of type 'ParamListInfo' {aka 'struct ParamListInfoData *'}
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                              ^~~~~~
pgtt.c:373:45: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c:98:70: warning: passing argument 5 of 'prev_ProcessUtility' from incompatible pointer type [-Wincompatible-pointer-types]
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                      ^~~~~~~~
      |                                                                      |
      |                                                                      QueryEnvironment *
pgtt.c:373:45: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c:98:70: note: expected 'ParamListInfo' {aka 'struct ParamListInfoData *'} but argument is of type 'QueryEnvironment *'
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                      ^~~~~~~~
pgtt.c:373:45: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c:98:80: warning: passing argument 6 of 'prev_ProcessUtility' from incompatible pointer type [-Wincompatible-pointer-types]
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                                ^~~~
      |                                                                                |
      |                                                                                DestReceiver * {aka struct _DestReceiver *}
pgtt.c:373:45: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c:98:80: note: expected 'QueryEnvironment *' but argument is of type 'DestReceiver *' {aka 'struct _DestReceiver *'}
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                                ^~~~
pgtt.c:373:45: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c:98:86: warning: passing argument 7 of 'prev_ProcessUtility' from incompatible pointer type [-Wincompatible-pointer-types]
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                                      ^~
      |                                                                                      |
      |                                                                                      QueryCompletion *
pgtt.c:373:45: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c:98:86: note: expected 'DestReceiver *' {aka 'struct _DestReceiver *'} but argument is of type 'QueryCompletion *'
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                                      ^~
pgtt.c:373:45: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c:373:25: error: too few arguments to function 'prev_ProcessUtility'
  373 |                         prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                         ^~~~~~~~~~~~~~~~~~~
pgtt.c:98:62: error: incompatible type for argument 4 of 'standard_ProcessUtility'
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                              ^~~~~~
      |                                                              |
      |                                                              ParamListInfo {aka struct ParamListInfoData *}
pgtt.c:375:49: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  375 |                         standard_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~
In file included from pgtt.c:51:
/usr/pgsql-14/include/server/tcop/utility.h:87:95: note: expected 'ProcessUtilityContext' but argument is of type 'ParamListInfo' {aka 'struct ParamListInfoData *'}
   87 |                                                                         ProcessUtilityContext context, ParamListInfo params,
      |                                                                         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
pgtt.c:98:70: warning: passing argument 5 of 'standard_ProcessUtility' from incompatible pointer type [-Wincompatible-pointer-types]
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                      ^~~~~~~~
      |                                                                      |
      |                                                                      QueryEnvironment *
pgtt.c:375:49: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  375 |                         standard_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~
In file included from pgtt.c:51:
/usr/pgsql-14/include/server/tcop/utility.h:87:118: note: expected 'ParamListInfo' {aka 'struct ParamListInfoData *'} but argument is of type 'QueryEnvironment *'
   87 |                                                                         ProcessUtilityContext context, ParamListInfo params,
      |                                                                                                        ~~~~~~~~~~~~~~^~~~~~
pgtt.c:98:80: warning: passing argument 6 of 'standard_ProcessUtility' from incompatible pointer type [-Wincompatible-pointer-types]
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                                ^~~~
      |                                                                                |
      |                                                                                DestReceiver * {aka struct _DestReceiver *}
pgtt.c:375:49: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  375 |                         standard_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~
In file included from pgtt.c:51:
/usr/pgsql-14/include/server/tcop/utility.h:88:91: note: expected 'QueryEnvironment *' but argument is of type 'DestReceiver *' {aka 'struct _DestReceiver *'}
   88 |                                                                         QueryEnvironment *queryEnv,
      |                                                                         ~~~~~~~~~~~~~~~~~~^~~~~~~~
pgtt.c:98:86: warning: passing argument 7 of 'standard_ProcessUtility' from incompatible pointer type [-Wincompatible-pointer-types]
   98 | #define GTT_PROCESSUTILITY_ARGS pstmt, queryString, context, params, queryEnv, dest, qc
      |                                                                                      ^~
      |                                                                                      |
      |                                                                                      QueryCompletion *
pgtt.c:375:49: note: in expansion of macro 'GTT_PROCESSUTILITY_ARGS'
  375 |                         standard_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~
In file included from pgtt.c:51:
/usr/pgsql-14/include/server/tcop/utility.h:89:87: note: expected 'DestReceiver *' {aka 'struct _DestReceiver *'} but argument is of type 'QueryCompletion *'
   89 |                                                                         DestReceiver *dest, QueryCompletion *qc);
      |                                                                         ~~~~~~~~~~~~~~^~~~
pgtt.c:375:25: error: too few arguments to function 'standard_ProcessUtility'
  375 |                         standard_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~
In file included from pgtt.c:51:
/usr/pgsql-14/include/server/tcop/utility.h:85:13: note: declared here
   85 | extern void standard_ProcessUtility(PlannedStmt *pstmt, const char *queryString,
      |             ^~~~~~~~~~~~~~~~~~~~~~~
pgtt.c: In function 'create_temporary_table_internal':
pgtt.c:1569:66: error: incompatible type for argument 4 of 'ProcessUtility'
 1569 |                                                                  NULL, NULL,
      |                                                                  ^~~~
      |                                                                  |
      |                                                                  void *
In file included from pgtt.c:51:
/usr/pgsql-14/include/server/tcop/utility.h:82:74: note: expected 'ProcessUtilityContext' but argument is of type 'void *'
   82 |                                                    ProcessUtilityContext context, ParamListInfo params,
      |                                                    ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
pgtt.c:1570:66: warning: passing argument 6 of 'ProcessUtility' from incompatible pointer type [-Wincompatible-pointer-types]
 1570 |                                                                  None_Receiver,
      |                                                                  ^~~~~~~~~~~~~
      |                                                                  |
      |                                                                  DestReceiver * {aka struct _DestReceiver *}
In file included from pgtt.c:51:
/usr/pgsql-14/include/server/tcop/utility.h:83:70: note: expected 'QueryEnvironment *' but argument is of type 'DestReceiver *' {aka 'struct _DestReceiver *'}
   83 |                                                    QueryEnvironment *queryEnv,
      |                                                    ~~~~~~~~~~~~~~~~~~^~~~~~~~
pgtt.c:1566:25: error: too few arguments to function 'ProcessUtility'
 1566 |                         ProcessUtility(stmt,
      |                         ^~~~~~~~~~~~~~
In file included from pgtt.c:51:
/usr/pgsql-14/include/server/tcop/utility.h:80:13: note: declared here
   80 | extern void ProcessUtility(PlannedStmt *pstmt, const char *queryString,
      |             ^~~~~~~~~~~~~~
make[1]: *** [<builtin>: pgtt.o] Error 1
make[1]: *** Waiting for unfinished jobs....
pgtt.c:304:22: warning: incompatible function pointer types assigning to 'ProcessUtility_hook_type' (aka 'void (*)(struct PlannedStmt *, const char *, bool, ProcessUtilityContext, struct ParamListInfoData *, struct QueryEnvironment *, struct _DestReceiver *, struct QueryCompletion *)') from 'void (PlannedStmt *, const char *, ProcessUtilityContext, ParamListInfo, QueryEnvironment *, DestReceiver *, QueryCompletion *)' (aka 'void (struct PlannedStmt *, const char *, ProcessUtilityContext, struct ParamListInfoData *, struct QueryEnvironment *, struct _DestReceiver *, struct QueryCompletion *)') [-Wincompatible-function-pointer-types]
        ProcessUtility_hook = gtt_ProcessUtility;
                            ^ ~~~~~~~~~~~~~~~~~~
pgtt.c:373:47: error: too few arguments to function call, expected 8, have 7
                        prev_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
                        ~~~~~~~~~~~~~~~~~~~                        ^
pgtt.c:375:51: error: too few arguments to function call, expected 8, have 7
                        standard_ProcessUtility(GTT_PROCESSUTILITY_ARGS);
                        ~~~~~~~~~~~~~~~~~~~~~~~                        ^
/usr/pgsql-14/include/server/tcop/utility.h:85:13: note: 'standard_ProcessUtility' declared here
extern void standard_ProcessUtility(PlannedStmt *pstmt, const char *queryString,
            ^
pgtt.c:1571:14: error: too few arguments to function call, expected 8, have 7
                                                                 NULL);
                                                                     ^
/usr/pgsql-14/include/server/tcop/utility.h:80:13: note: 'ProcessUtility' declared here
extern void ProcessUtility(PlannedStmt *pstmt, const char *queryString,
            ^
1 warning and 3 errors generated.
make[1]: *** [/usr/pgsql-14/lib/pgxs/src/makefiles/../../src/Makefile.global:1070: pgtt.bc] Error 1
make[1]: Leaving directory '/home/devrim/Documents/Devrim/Projects/repo/pgrpms/rpm/redhat/master/non-common/pgtt/main/pgtt-2.5'
error: Bad exit status from /var/tmp/rpm-tmp.RvIuYJ (%build)

**Comment by darold** [comment]
Thanks for the remainder Devrim, this is now fixed and a new release 2.6 has been publishedt.

**Comment by devrimgunduz** [comment]
Thanks a lot Gilles!
